benchmarks/mbpp/mbpp.py

```diff
@@ -0,0 +1,164 @@
+"""
+MBPP: Mostly Basic Python Problems
+
+Jacob Austin, Augustus Odena, Maxwell Nye, Maarten Bosma, Henryk Michalewski,
+David Dohan, Ellen Jiang, Carrie Cai, Michael Terry, Quoc Le, Charles Sutton
+https://arxiv.org/pdf/2108.07732
+
+Based on: https://github.com/google-research/google-research/tree/master/mbpp
+
+# run the eval on the sanitized test split
+inspect eval benchmarks/mbpp/mbpp.py
+"""
+import math  # Used by some assert statements by MBPP.
+
+from datasets import load_dataset
+
+from inspect_ai import Task, task
+from inspect_ai.dataset import Sample, hf_dataset
+from inspect_ai.model import GenerateConfig
+from inspect_ai.scorer import (CORRECT, INCORRECT, Score, Target, accuracy,
+                               scorer, stderr)
+from inspect_ai.solver import TaskState, generate, prompt_template
+
+PROMPT_TEMPLATE = """
+You are an expert Python programmer. You will be given a task, and the tests that your code must pass.
+Write the Python function to solve the task. Do not give additional explanations, just output the
+Python function. Only use imports that are included in Python's standard library.
+"""
+
+
+def record_to_sample(record):
+    return Sample(
+        input=record["prompt"],
+        target=record["test_list"],
+        id=record["task_id"],
+        metadata={
+            "prompt": record["prompt"],
+            "test_list": record["test_list"],
+            "test_list_str": "\n".join(record["test_list"]),
+            "source_file": record["source_file"],
+            "code": record["code"],
+            "test_imports": record["test_imports"],
+            "task_id": record["task_id"],
+        }
+    )
+
+
+@scorer(metrics=[accuracy(), stderr()])
+def code_acceptance():
+    async def score(state: TaskState, target: Target):
+
+        # It is assumed that generated output is of the form:
+        # ```python
+        # [code output]
+        # ```
+        raw_generated_code = state.output.completion
+        generated_code = raw_generated_code.replace("```python", "").replace("```", "")
+
+        explanation = ""
+        passed = True
+        try:
+            # Pass globals() as exec() may not work for functions that are recursive
+            # or use other functions that will still be defined afterwards.
+            exec(generated_code, globals())
```

Comment:
We need to execute this code in a secure sandbox--you can see how we do this in the HumanEval benchmark: https://github.com/UKGovernmentBEIS/inspect_ai/blob/main/benchmarks/humaneval/humaneval.py#L79

Note we should further specify `sandbox="docker"` in the Task: https://github.com/UKGovernmentBEIS/inspect_ai/blob/main/benchmarks/humaneval/humaneval.py#L61 (note if you don't have Docker for development you can use `sandbox="local"`.

If you need to copy files to and from the sandbox here is the API for that: https://inspect.ai-safety-institute.org.uk/agents.html#sec-sandbox-environments

Reply:
This should be fixed in this PR:

https://github.com/UKGovernmentBEIS/inspect_ai/pull/356

Thanks for catching this one!